src/odemis/acq/move.py

```diff
@@ -52,685 +55,728 @@
+        :param component: Either the stage or the align component
+        :raises CancelledError: if the reference is cancelled
+        """
+        try:
+            with future._task_lock:
+                if future._task_state == CANCELLED:
+                    logging.info("Reference procedure is cancelled.")
+                    raise CancelledError()
+                logging.debug("Performing stage referencing.")
+                future._running_subf = component.reference(set(component.axes.keys()))
+            future._running_subf.result()
+        except Exception as error:
+            logging.exception(error)
+        if future._task_state == CANCELLED:
+            logging.info("Reference procedure is cancelled.")
+            raise CancelledError()
+
+    def _run_sub_move(self, future, component, sub_move):
+        """
+        Perform the sub moveAbs using the given component and axis->pos dict
+        :param future: cancellable future of the whole move
+        :param component: Either the stage or the align component
+        :param sub_move: the sub_move axis->pos dict
+        :raises TimeoutError: if the sub move timed out
+        :raises CancelledError: if the sub move is cancelled
+        """
+        try:
+            with future._task_lock:
+                if future._task_state == CANCELLED:
+                    logging.info("Move procedure is cancelled before moving %s -> %s", component.name, sub_move)
+                    raise CancelledError()
+
+                logging.debug("Performing sub move %s -> %s", component.name, sub_move)
+                future._running_subf = component.moveAbs(sub_move)
+            future._running_subf.result(timeout=MAX_SUBMOVE_DURATION)
+        except TimeoutError:
+            future._running_subf.cancel()
+            logging.exception("Timed out while moving %s -> %s", component.name, sub_move)
+            raise
+
+        if future._task_state == CANCELLED:
+            logging.info("Move procedure is cancelled after moving %s -> %s", component.name, sub_move)
+            raise CancelledError()
+
+
+class MeteorPostureManager(MicroscopePostureManager):
+    def __init__(self, microscope):
+        # Load components
+        self.stage = model.getComponent(role='stage-bare')
+        self.focus = model.getComponent(role='focus')
+        # set linear axes and rotational axes used
+        self.axes = self.stage.axes
+        self.linear_axes = set(key for key in self.axes.keys() if key in {'x', 'y', 'z', 'm'})
+        self.rotational_axes = set(key for key in self.axes.keys() if key in {'rx', 'ry', 'rz', 'rm'})
+        # required keys that must be present in the stage metadata
+        self.required_keys = {
+            model.MD_FAV_POS_DEACTIVE, model.MD_FAV_SEM_POS_ACTIVE, model.MD_FAV_FM_POS_ACTIVE,
+            model.MD_SAMPLE_CENTERS}
+
+    def getCurrentPostureLabel(self, pos: Dict[str, float]) -> int:
```

Comment:
Please drop the `pos` argument as well. You can read it (now) directly with `self.stage.position.value`. 

I don't think so, but if there is some caller that requires the name of different position than the current one, then you can leave this argument, but only as optional (and typically, it'd be `None`, in which case the position is read from self.stage).